src/main/java/uk/co/ramp/covid/simulation/population/Person.java

```diff
@@ -358,4 +367,85 @@ public void unsetMoved() {
     protected abstract double getInfectionSeedRate();
     
     public void furlough() {};
+
+    // We can't determine this in household in case the person is working nightshift
+    // Time is always the start of a day
+    public void deteremineHospitalVisits(Time t, boolean lockdown, Places places) {
+        HospitalApptInfo info = PopulationParameters.get().hospitalAppsParams().getParams(sex, age);
+        
+        // Appts might be across days so don't regenerate if we already have one
+        if (hasHospitalAppt() && !getHospitalAppt().isOver(t)) {
+            return;
+        }
+        
+        double lockdownAdjust = lockdown ? 0.75 : 1.0;
+
+        if (new Probability(info.pInPatient.asDouble() * lockdownAdjust).sample()) {
+            
+            Time startTime = new Time(t.getAbsTime() +
+                    RNG.get().nextInt(
+                            PopulationParameters.get().hospitalApptTimings.inPatientFirstStartTime,
+                            PopulationParameters.get().hospitalApptTimings.inPatientLastStartTime));
+
+            int length = info.inPatientLengthDays.intValue() * 24;
+            if (length < 1) {
```

Comment:
This looks like an invalid parameter, rather than something to check at runtime?

Reply:
However, what I think you've actually got too much copy and pasting going on here, and want something more like:
```
        Time startTime = null;
        int length;

        if (sampleHospitalVisitProbability(info.pInPatient, lockdown)) {
            
            startTime = new Time(t.getAbsTime() +
                    RNG.get().nextInt(
                            PopulationParameters.get().hospitalApptTimings.inPatientFirstStartTime,
                            PopulationParameters.get().hospitalApptTimings.inPatientLastStartTime));

            length = info.inPatientLengthDays.intValue() * 24;
           
        } else if (sampleHospitalVisitProbability(info.pDayCase, lockdown)) {

            startTime = new Time(t.getAbsTime() +
                    PopulationParameters.get().hospitalApptTimings.dayCaseStartTime);

            length = (int) RNG.get().nextGaussian(
                    PopulationParameters.get().hospitalApptTimings.meanDayCaseTime,
                    PopulationParameters.get().hospitalApptTimings.SDDayCaseTime
            );
            
        } else if (sampleHospitalVisitProbability(info.pOutPatient, lockdown)) {
...
        }

        hospitalAppt = null;

        if (startTime != null) {
            if (length < 1) { length =  1; }

            // For small populations there might not be any non-COVID hospitals
            Hospital h = places.getRandomNonCovidHospital();
            if (h != null) {
                hospitalAppt = new HospitalAppt(startTime, length, h);
            }
        }

```
with
```
private boolean sampleHospitalVisitProbability(Probability prob, boolean lockdown) {
    if (lockdown) {
        return Probability.sample(prob.asDouble() * PopulationParameters.get().hospitalApptProbabilityLockdownAdjustment);
    } else {
        return prob.sample();
    }
}
```